docs/state.rst

```diff
@@ -244,3 +246,119 @@ For example:
         otherAppOtherAccountRole,
         If(otherAppOtherAccountRole.hasValue(), otherAppOtherAccountRole.value(), Bytes("none"))
     ])
+
+Box Storage
+-----------
+
+Box storage consists of key-value pairs that are stored in application's local context.
+
+The app account's minimum balance requirement (MBR) is increased by each box, and by the byte size of the box.
+If one deletes an application with outstanding boxes, the MBR is not recoverable from the deleted app account.
+
+Box sizes and names cannot be changed after initial allocation, but they can be deleted and re-allocated.
+Boxes are only visible to the app itself, other apps cannot view the app's own boxes on-chain.
+
+Boxes can be manipulated as follows:
+
+Create Boxes
+~~~~~~~~~~~~
+
+To create a box, use :any:`App.box_create`, or :any:`App.box_put` method.
+
+For :any:`App.box_create`, the first argument is the box name, and the second argument is the byte size to be allocated.
+
+For example:
+
+.. code-block:: python
+
+    # Allocate a box called "BoxForAlice" of byte size 100
+    App.box_create(Bytes("BoxForAlice"), Int(0xA11CE))
+    # Allocate a box called "BoxForBob" of byte size 90
+    App.box_create(Bytes("BoxForBob"), Int(0xB0B)
+
+For :any:`App.box_put`, the first argument is the box name to create or to write to, and the second argument is the bytes to write.
+
+**Notice**: if the box exists, then :any:`App.box_put` will write the contents to the box;
+otherwise, it will create a box containing exactly the same input bytes.
+
+.. code-block:: python
+
+    # create a 42 bytes length box called `poemLine` with content
+    App.box_put(Bytes("poemLine"), Bytes("Of that colossal wreck, boundless and bare"))
+    # write to box `poemLine` with new value
+    App.box_put(Bytes("poemLine"), Bytes("The lone and level sands stretch far away."))
+
+Writing to a Box
+~~~~~~~~~~~~~~~~
+
+To write to a box, use :any:`App.box_replace`, or :any:`App.box_put` method.
+
+For :any:`App.box_replace`, it writes bytes of certain length from a start index in a Box.
+The first argument is the box name to write into, the second argument is the starting index to write,
+and the third argument is the replacement bytes. For example:
+
+.. code-block:: python
+
+   # replace 2 bytes to `Ne` starting from 0st bytes in a box with name `wordleBox`
+   App.box_replace(Bytes("wordleBox"), Int(0), Bytes("Ne"))
+
+For :any:`App.box_put`, it writes the full content to a pre-existing box.
```

Comment:
```suggestion
:any:`App.box_put` writes the full contents to a pre-existing box.
```

Reply:
nit: maybe this section is a bit redundant given the example for `box_put` previously provided? But it's also arguable to keep this example given that this section is about "writing". A middle ground would be to just mention that ":any:`App.box_put` can also be used for writing a box'es entire contents, as seen in the example above."

Reply:
indeed, it is redundant for `box_put`, as it was previously stated. I was thinking of a way to circumvent re-writing same paragraph cross different sections.

The middle ground suggestion makes sense to me, I will adapt to that suggestion and reflect in the document.